bastion.cfn.yml

```diff
@@ -16,68 +16,290 @@ Parameters:
 
   KeyName:
     Description: EC2 key pair name for bastion host SSH access
+    Type: AWS::EC2::KeyPair::KeyName
+
+  LogRetentionInDays:
+    Description: Number of days you would like you cloudwatch logs to be retained
+    Type: Number
+    Default: 7
+
+  MFA:
+    Description: Change it to true to install MFA using Google Authenticator on your instance
     Type: String
-    MinLength: 1
-    MaxLength: 255
-    AllowedPattern: "[\\x20-\\x7E]*"
-    ConstraintDescription: Key pair name can contain only ASCII characters.
+    ConstraintDescription: Value must be true or false
+    Default: false
+    AllowedValues:
+      - true
+      - false
 
 Mappings:
 
   AMIMap:  # Amazon Linux AMI
```

Comment:
A comment about where to find these AMI IDs might be helpful to a reader. Looking at the current AMI IDs published on aws.amazon.com, these seem out of date already.